frontend/src/metabase/visualizations/components/settings/ChartSettingInputs.jsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+import Input from "metabase/components/Input.jsx";
+
+// value is an array of strings. This component provides one input box per string
+export default function ChartSettingInputs({ value, onChange }) {
+    const inputs = value.map((str, i) => (
+        <Input
+            key={i}
+            className="input block full"
+            value={str}
+            onBlurChange={(e) => {
+                const newStr = e.target.value.trim();
+                if (!newStr || !newStr.length) return;
+
+                const newValue = value.slice(); // clone the original `value` array
+                newValue[i] = newStr;
+                onChange(newValue);
```

Comment:
Something about this having to clone the array, update a string at an index, and then return the whole array seems a little complicated. I'm guessing you're doing it this way cause of how the existing settings framework pairs widgets with certain values and there's not really a place to put this update logic other than here?

Reply:
Yeah unfortunately the array is part of state or something so it's read-only. 

That's how the other widgets in the Settings framework work FWIW (define a setting in `graph.js` and a widget that handles editing that setting).

I did manage to make this general (i.e., not specific to changing series titles) so it could be used for other arrays of strings in the future. At least we have that

But yes I agree the Settings framework relies on a lot of magic so everything has to be done in a weird sort of way